multiversion/src/main/scala/multiversion/configs/ThirdpartyConfig.scala

```diff
@@ -81,10 +81,15 @@ final case class ThirdpartyConfig(
       cdep: Dependency
   ): Result[Task[Result[DependencyResolution]]] =
     Result.fromResults(decodeForceVersions(dep)).map { forceVersions =>
+      // The dependencies on other 3rdparty targets should be resolved together.
+      val targetDeps = for {
+        dependency <- dep.dependencies
+        dep <- depsByTargets.getOrElse(dependency, Nil)
+      } yield dep.toCoursierDependency(scala)
```

Comment:
I don't think this would work because Pants BUILD files would often include exclusion clause to prioritize the 3rdparty resolution, so let's say you got com.google.api.grpc:proto-google-cloud-bigtable-v2:0.19.0 with exclude clause that excludes Guava, adding Guava into the mock module is not going to make Guava get resolved because it's excluded.

Note we _cannot_ ignore excludes either because sometimes we do want to actually exclude things too. This is why in the ticket I proposed add the `scala_import` reference to what it thinks `3rdparty/jvm/com/google/guava` would point to instead.

Reply:
> adding Guava into the mock module is not going to make Guava get resolved because it's excluded.

The guava version that would come as part of the resolution of `com.google.api.grpc:proto-google-cloud-bigtable-v2:0.19.0` will be excluded, and replaced with the guava version that is included in the mock module.

Setting multiple root dependencies to resolve matches the current implementation of coursier resolution in pants: https://github.com/pantsbuild/pants/blob/1.25.x-twtr/src/python/pants/backend/jvm/tasks/coursier_resolve.py#L469-L522